atlp-rwanda/eagles-ec-be
# #187419174 User should be able to update their password

#### What does this PR do?
This PR implements update password feature

#### Description of Task to be completed?
   - Implement the feature to allow users update their password
   - Added validation check for old password, new password and confirm password
   - hash the new password before updating
   - documented the feature using swagger

#### How should this be manually tested?
1. Clone the repository.
2. Checkout to the branch `ft-passwordupdate-#187419174`.
3. Run `npm install` to install dependencies.
4. Run `npm run dev` to start the development server.
5. Open a web browser and navigate to the `/docs` endpoint (e.g., `http://localhost:PORT/docs`).
6. Verify that the Swagger UI loads correctly.
7. First authorize by logging in any user and use the token provided
8. Explore the endpoint PUT on  `/api/v1/users/passwordupdate` and verify that updating password is working.

#### What are the relevant pivotal tracker stories?

[Delivers #187419174]
